src/main/java/net/onelitefeather/bettergopaint/listeners/InteractListener.java

```diff
@@ -18,97 +18,100 @@
         }
-        if ((e.getAction().equals(Action.RIGHT_CLICK_AIR) || e.getAction().equals(Action.RIGHT_CLICK_BLOCK))) {
-            if (e.getPlayer().getItemInHand().hasItemMeta() && e.getPlayer().getItemInHand().getItemMeta().hasDisplayName() && e
-                    .getPlayer()
-                    .getItemInHand()
-                    .getItemMeta()
-                    .getDisplayName()
-                    .startsWith(" <aqua>♦ ") && e.getPlayer().getItemInHand().getItemMeta().hasLore()) {
-                final ExportedPlayerBrush epb = new ExportedPlayerBrush(e
-                        .getPlayer()
-                        .getItemInHand()
-                        .getItemMeta()
-                        .getDisplayName(), e.getPlayer().getItemInHand().getItemMeta().getLore());
-                final Player p = e.getPlayer();
-                final Location loc;
-                if (e.getAction().equals(Action.RIGHT_CLICK_AIR)) {
-                    loc = p.getTargetBlock(null, 250).getLocation().clone();
-                } else {
-                    loc = e.getClickedBlock().getLocation().clone();
-                }
-                BukkitAdapter.adapt(p).runAsyncIfFree(() ->epb.getBrush().paint(loc, p, epb));
-            }
+
+        if (event.getAction().isLeftClick() && item.getType().equals(Material.FEATHER)) {
+            PlayerBrush brush = plugin.getBrushManager().getBrush(player);
+            player.openInventory(brush.getInventory());
+            event.setCancelled(true);
+            return;
         }
-        if (e.getPlayer().getItemInHand().getType() == XMaterial.FEATHER.parseMaterial() && (e
-                .getAction()
-                .equals(Action.RIGHT_CLICK_AIR) || e.getAction().equals(Action.RIGHT_CLICK_BLOCK))) {
-            e.setCancelled(true);
-            final Player p = e.getPlayer();
-            final Location loc;
-            if (e.getAction().equals(Action.RIGHT_CLICK_AIR)) {
-                loc = p.getTargetBlock(null, 250).getLocation().clone();
-            } else {
-                loc = e.getClickedBlock().getLocation().clone();
-            }
-            if ((!e.getPlayer().hasPermission("bettergopaint.world.bypass")) && (Settings.settings().GENERIC.DISABLED_WORLDS
-                    .contains(loc.getWorld().getName()))) {
-                return;
-            }
-            if (loc.getBlock().getType().equals(XMaterial.AIR.parseMaterial())) {
-                return;
-            }
-            final PlayerBrush pb = BetterGoPaint.getBrushManager().getPlayerBrush(p);
-
-            if (pb.isEnabled()) {
-                BukkitAdapter.adapt(p).runAsyncIfFree(() -> pb.getBrush().paint(loc, p));
-            } else {
-                p.sendMessage(MiniMessage.miniMessage().deserialize(Settings.settings().GENERIC.PREFIX + "<red>Your brush is " +
-                        "disabled, left click to enable the brush."));
-            }
+
+        if (!event.getAction().isRightClick()) {
```

Comment:
Extract into extra variable for better readability

Reply:
what exactly?